rules/ruleset.go

```diff
@@ -71,84 +69,115 @@ func newActiveRuleSet(
 	uniqueCutoverTimes := make(map[int64]struct{})
 	for _, mappingRule := range mappingRules {
 		for _, snapshot := range mappingRule.snapshots {
-			uniqueCutoverTimes[snapshot.cutoverNs] = struct{}{}
+			uniqueCutoverTimes[snapshot.cutoverNanos] = struct{}{}
 		}
 	}
 	for _, rollupRule := range rollupRules {
 		for _, snapshot := range rollupRule.snapshots {
-			uniqueCutoverTimes[snapshot.cutoverNs] = struct{}{}
+			uniqueCutoverTimes[snapshot.cutoverNanos] = struct{}{}
 		}
 	}
 
-	cutoverTimeAsc := make([]int64, 0, len(uniqueCutoverTimes))
+	cutoverTimesAsc := make([]int64, 0, len(uniqueCutoverTimes))
 	for t := range uniqueCutoverTimes {
-		cutoverTimeAsc = append(cutoverTimeAsc, t)
+		cutoverTimesAsc = append(cutoverTimesAsc, t)
 	}
-	sort.Sort(int64Asc(cutoverTimeAsc))
+	sort.Sort(int64Asc(cutoverTimesAsc))
 
 	return &activeRuleSet{
-		version:         version,
 		iterFn:          iterFn,
 		newIDFn:         newIDFn,
 		mappingRules:    mappingRules,
 		rollupRules:     rollupRules,
-		cutoverTimesAsc: cutoverTimeAsc,
+		cutoverTimesAsc: cutoverTimesAsc,
 	}
 }
 
-func (as *activeRuleSet) Match(id []byte, t time.Time) MatchResult {
-	timeNs := t.UnixNano()
-	mappingCutoverNs, mappingPolicies := as.matchMappings(id, timeNs)
-	rollupCutoverNs, rollupResults := as.matchRollups(id, timeNs)
+// NB(xichen): could make this more efficient by keeping track of matched rules
+// at previous iteration and incrementally update match results.
+func (as *activeRuleSet) MatchAll(id []byte, from time.Time, to time.Time) MatchResult {
```

Comment:
Just wondering about our interface/contracts here.  Do we want most to be in unix nanoseconds as int64 or as `time.Time`?  

I think we can chop and change, but we should be mindful about which we pick and where.

As a nit: can save a `time.Time` declaration here using `from, to time.Time`.